R/dependencies.R

```diff
@@ -242,8 +242,32 @@ renv_dependencies_callback <- function(path) {
     ".rnw"         = function(path) renv_dependencies_discover_multimode(path, "rnw")
   )
 
-  cbname[[basename(path)]] %||% cbext[[tolower(fileext(path))]]
+  cbname[[basename(path)]] %||% cbext[[tolower(fileext(path))]] %||% {
+    if (is_r_executable(path)) function(path) renv_dependencies_discover_r(path)
+  }
+
+}
 
+is_r_executable <- function(path) {
+  if (nzchar(fileext(path)))
+    return(FALSE)
+
+  grepl("\\b(R|r|Rscript)\\b", get_shebang_command(path))
```

Comment:
Could we add a comment here explaining why we don't check if the file is executable (ie, for Windows)?

Reply:
How about now?